### PR TITLE
Adding Kuberenetes templates and deployment guide

### PR DESCRIPTION
**Pull request by Langhalsdino**
### Motivation and context
The topic was raised a couple of times in issues like #1087 . Since kubernetes is widely use easy deployment into the kubernetes environment would provide great value to the community and help to get cvat to a wider audience. 

Special due to changes like #1641 its now way easier to deploy cvat in a k8s environment.

### How has this been tested?
I deployed this in a couple of namespaces with in our cluster (with and without nvida gpu). 
Furthermore i did not do any changes to the code, therefore the only real issue was networking. Since i was following the docker-compose.yml closely there where no real challges

### Checklist

- [X] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [X] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [X] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [X] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by ActiveChooN** [comment]
Hi @Langhalsdino, thanks a lot for your contribution. In our team we think a lot about adding kubernetes or docker swarm support for deployment. I have a few questions and suggestions.

1) We recently merged PR (#1767) with DL models serving as a function with [Nuclio](https://github.com/nuclio/nuclio/). Do you want to add this in your templates?
2) There is a nice way to organize the kubernetes structure with [Helm](https://github.com/helm/helm) charts and [helmfile](https://github.com/roboll/helmfile). This can help to separate configuration values from templates itself and provides easy way to deploy and manage application. What about to use this tools in your PR?

**Comment by Langhalsdino** [comment]
> There is a nice way to organize the kubernetes structure with Helm charts and helmfile. This can help to separate configuration values from templates itself and provides easy way to deploy and manage application. What about to use this tools in your PR?

Building helm files is in general a good idea and would help to set all the variable such as docker image registry, domains, ... 
I will focus on this next weekend and try to add it with some new commits.

> We recently merged PR (#1767) with DL models serving as a function with Nuclio. Do you want to add this in your templates?

I could add it to the template. I could not find any examples on how to use nuclio, do you have example models, ... such that i could test the deployment.

Lastly do you know the current status of #7 , since referencing official images in the deployment templates makes it way clearer and could  prepare the ground for publishing cvat as an official helm chart on helm.sh

In general i am not a big fan of having growing pull requests. Therefore i would prefere if we could agree on a certain scope of this issue and maybe split it into multiple once, such that we will see them merged :)

**Comment by nmanovic** [comment]
@Langhalsdino , a couple of examples with nuclio you can find here: https://github.com/opencv/cvat/blob/develop/serverless/deploy.sh

What is the reasonable scope for the PR from your point of view?

**Comment by Langhalsdino** [comment]
> What is the reasonable scope for the PR from your point of view?

I am tempted to split this into two issues and PR. 
The first one is adding a helm chart for the basic setup of CVAT and second one would be extending it with a nuclio container (incl. GPU support).

Basic helm chart would include a statefull set of:
- PostgresDB
- Redis
- CVAT
- CVAT-UI 
- Service

The other issue would extend the functionality by adding nuclio support and GPU support for CVAT and Nuclio.

**Comment by rushtehrani** [comment]
If the plan is to use Helm, note that Nuclio supports and recommends the [Helm chart deployment](https://nuclio.io/docs/latest/setup/k8s/running-in-production-k8s/#the-preferred-deployment-method) for production.

GPU support is dependent on the provider, for example GKE automatically installs the [NVIDIA device plugin daemonset](https://github.com/NVIDIA/k8s-device-plugin) but AKS and EKS require you to install it manually.

**Comment by Langhalsdino** [comment]
> GPU support is dependent on the provider, for example GKE automatically installs the NVIDIA device plugin daemonset but AKS and EKS require you to install it manually.

I know about the plugin deamonset and i would mention it in a Readme, such that every EKS and AKS user will read it after there pods are not being scheduled :D 

> If the plan is to use Helm, note that Nuclio supports and recommends the Helm chart deployment for production.
Thats looks awesome. Then we will not need to split this into two issues and i will just link to their helm repo and add configuration options for connecting to Nuclio.

> Also, now that inference is in Nuclio, do we still need to have GPU support in the main CVAT container?
This will probably make it a lot easier. If the roadmap of CVAT is pursuing a deeper integration of nuclio, i am happy to skipp the GPU part. Since it is different for every platform 🙈

**Comment by rushtehrani** [comment]
That all sounds great! At some point we may also want to support a [Kustomize](https://kustomize.io/) deployment too.

**Comment by Langhalsdino** [comment]
> That all sounds great! At some point we may also want to support a Kustomize deployment too.

Why do you want to go for Kustomize, if there is already a helm chart? I can not see the benefits of having both.

**Comment by rushtehrani** [comment]
> Why do you want to go for Kustomize, if there is already a helm chart? I can not see the benefits of having both.
> If kustomize ist the way to go, i could take a look into it and build it instead of helm. Eventhough helm is to be very popular and i use it on a daily bases.

To clarify, I think going with Helm for this phase makes perfect sense.

Ideally we'd want to support both. Kustomize is built into `kubectl` and projects like ours and [Kubeflow](https://www.kubeflow.org/docs/started/getting-started/) use it for mainly the reasons outlined in [this overview](https://kustomize.io/#overview).

We can contribute the Kustomize version at a later time.

**Comment by nmanovic** [comment]
> > What is the reasonable scope for the PR from your point of view?
> 
> I am tempted to split this into two issues and PR.
> The first one is adding a helm chart for the basic setup of CVAT and second one would be extending it with a nuclio container (incl. GPU support).
> 
> Basic helm chart would include a statefull set of:
> 
> * PostgresDB
> * Redis
> * CVAT
> * CVAT-UI
> * Service
> 
> The other issue would extend the functionality by adding nuclio support and GPU support for CVAT and Nuclio.
> 
> What do you think about it @nmanovic

Looks great! Thanks for the contribution.

File: kubernetes-templates/README.md
Comment (aaniin) [inline kubernetes-templates/README.md:18]: I get an error "docker build requires exactly 1 argument." I guess this should read
`docker build --file Dockerfile.ui \
  --tag $CI_REGISTRY_IMAGE/frontend:release-1.1.0 .` (without the dash) which succeeds in building for me.

Comment (Langhalsdino) [reply kubernetes-templates/README.md:18]: Thanks for pointing it out, i must have forgotten to remove the dash. I am used to adding the cache from option to every build, but did not want to commit it, to keep the template simple :) `--cache-from $CI_REGISTRY_IMAGE/frontend:release-1.1.0`

Comment (Langhalsdino) [reply kubernetes-templates/README.md]: Has been fixed in d1a855cf24a06f16812aaa3fd58530533116f8ed

**Comment by nmanovic** [comment]
@Langhalsdino , should we review the PR again or it is in progress? Could you please fix codacy issues?

**Comment by Langhalsdino** [comment]
Sorry i got lost in over engineering a helm chart and interrupted by upcoming exams and my thesis. I will fix codacy issues this weekend and open another PR with the helm chart later on.

**Comment by Langhalsdino** [comment]
@nmanovic I fixed the the issues mentioned in codacy. Regarding the CHANGE.log file. Should i add the changes in the following section or is this done by the repo maintainers?

```markdown
## [1.1.0-alpha] - 2020-06-30
### Added
```

**Comment by nmanovic** [comment]
@azhavoro , could you please look at the PR? Do you have any comments?

**Comment by nmanovic** [comment]
@Langhalsdino , sorry for the delay with our review and thanks for the contribution!

**Comment by jizg** [comment]
Hi, guys. I have tried deploying nuclio on my Minikube, and tested with CVAT deployed by @Langhalsdino 's YAML templates. The automatic annotation worked well in this way. I made some changes to the YAML templates, just created a service for nuclio in cvat namespace and pointed to nuclio-dashboard service in nuclio namespace, and added  no_proxy env for nuclio service to bypass cvat_proxy in `cvat_backend_deployment.yml`.

**Comment by Langhalsdino** [comment]
> Hi, guys. I have tried deploying nuclio on my Minikube, and tested with CVAT deployed by @Langhalsdino 's YAML templates. The automatic annotation worked well in this way. I made some changes to the YAML templates, just created a service for nuclio in cvat namespace and pointed to nuclio-dashboard service in nuclio namespace, and added no_proxy env for nuclio service to bypass cvat_proxy in `cvat_backend_deployment.yml`.

Could you share the confingurations

**Comment by Langhalsdino** [comment]
> Hi, guys. I have tried deploying nuclio on my Minikube, and tested with CVAT deployed by @Langhalsdino 's YAML templates. The automatic annotation worked well in this way. I made some changes to the YAML templates, just created a service for nuclio in cvat namespace and pointed to nuclio-dashboard service in nuclio namespace, and added no_proxy env for nuclio service to bypass cvat_proxy in `cvat_backend_deployment.yml`.

Could you share you changes, since i would like to include them in my template :)

**Comment by jizg** [comment]
Sure. Please check this PR https://github.com/apic-ai/cvat/pull/1
nuclio can be deployed based on the offical guide here (https://github.com/nuclio/nuclio/tree/development/docs/setup).

**Comment by nmanovic** [comment]
@azhavoro , could you please work together with @Langhalsdino to merge the PR? I see that CI is failed. But the functionality is really important for our internal infrastructure. Take care about it.

**Comment by Langhalsdino** [comment]
> Sure. Please check this PR [apic-ai#1](https://github.com/apic-ai/cvat/pull/1)
> nuclio can be deployed based on the offical guide here (https://github.com/nuclio/nuclio/tree/development/docs/setup).

@jizg thank you for the PR i merged it :)

**Comment by fg91** [comment]
Hey @Langhalsdino ,

thanks for the kubernetes templates, looking really forward to this functionality!

When trying, I find that the cvat backend cannot find redis.

Backend logs:

```
2020-11-11 15:45:14,309 DEBG 'rqscheduler' stderr output:
15:45:14 Registering birth

2020-11-11 15:45:14,339 DEBG 'rqscheduler' stderr output:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/redis/connection.py", line 559, in connect
    sock = self._connect()
  File "/usr/local/lib/python3.8/dist-packages/redis/connection.py", line 584, in _connect
    for res in socket.getaddrinfo(self.host, self.port, self.socket_type,
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/rqscheduler", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/rq_scheduler/scripts/rqscheduler.py", line 61, in main
    scheduler.run(burst=args.burst)
  File "/usr/local/lib/python3.8/dist-packages/rq_scheduler/scheduler.py", line 435, in run
    self.register_birth()
  File "/usr/local/lib/python3.8/dist-packages/rq_scheduler/scheduler.py", line 60, in register_birth
    if self.connection.exists(self.key) and \
  File "/usr/local/lib/python3.8/dist-packages/redis/client.py", line 1581, in exists
    return self.execute_command('EXISTS', *names)
  File "/usr/local/lib/python3.8/dist-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/usr/local/lib/python3.8/dist-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/usr/local/lib/python3.8/dist-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error -2 connecting to redis:6379. Name or service not known.

2020-11-11 15:45:14,355 DEBG fd 10 closed, stopped monitoring <POutputDispatcher at 139846789984064 for <Subprocess at 139846790140640 with name rqscheduler in state STARTING> (stdout)>
2020-11-11 15:45:14,355 DEBG fd 16 closed, stopped monitoring <POutputDispatcher at 139846789737680 for <Subprocess at 139846790140640 with name rqscheduler in state STARTING> (stderr)>
2020-11-11 15:45:14,355 INFO exited: rqscheduler (exit status 1; not expected)
2020-11-11 15:45:14,355 DEBG received SIGCHLD indicating a child quit
2020-11-11 15:45:15,357 INFO gave up: rqscheduler entered FATAL state, too many start retries too quickly
```

Do you know why this could be happening?
Thanks!

**Comment by Langhalsdino** [comment]
@nmanovic I am unable to view the snyk security output and therefore am not sure what failed. Could you share the output with me?

**Comment by Langhalsdino** [comment]
Hi @fg91 ,

since you send me an email regarding a different image i am assuming that you resolved the redis issue. 
If so, could you share what resolved the issue. If not i am happy to take a look into it.

In your email you raised the issue that hosting cvat on a specific path of a domain like`my.cool.domain/cvat` is not working. I have not tried this yet, but i would assume, that is currently not possible with cvat. Maybe a CVAT core developer can confirm or denial this.

**Comment by fg91** [comment]
> Hi @fg91 ,
> 
> since you send me an email regarding a different image i am assuming that you resolved the redis issue.
> If so, could you share what resolved the issue. If not i am happy to take a look into it.
> 
> In your email you raised the issue that hosting cvat on a specific path of a domain like`my.cool.domain/cvat` is not working. I have not tried this yet, but i would assume, that is currently not possible with cvat. Maybe a CVAT core developer can confirm or denial this.
> 
> If it is possible by cvat, i would investigate it further since it should be easily supported by adjusting the templates at a few places.

I replaced `"cvat-redis-service"` with `redis` as is hard-coded in 
https://github.com/openvinotoolkit/cvat/blob/b565721d81cd2357b82dd7cb9273e8eaf3b97222/supervisord.conf#L46

This is an ugly fix but avoids the problem for now.

**Comment by Langhalsdino** [comment]
> I replaced the service name `"cvat-redis-service"` with `redis` as is hard-coded in
> https://github.com/openvinotoolkit/cvat/blob/b565721d81cd2357b82dd7cb9273e8eaf3b97222/supervisord.conf#L46
> 
> This is an ugly fix but avoided the problem for me for now.

this is strange, since it should be handled by an env variable `CVAT_REDIS_HOST` that is set in the 04_cvat_backend_deployment.yml
````
          - name: CVAT_REDIS_HOST
            value: "cvat-redis-service"
```

**Comment by nmanovic** [comment]
@Langhalsdino , Try to merge develop. It should be resolved. Snyk complains abou lodash@4.17.19 module.

@azhavoro , we need the PR merged till EOM. Could you please look?

**Comment by nav13n** [comment]
@Langhalsdino  @nmanovic Is there any way I can help fast track this PR? Looking forward to using it.

**Comment by azhavoro** [comment]
> Hi @Langhalsdino @nmanovic, Is there any way I can help fast track this PR? Looking forward to using it.

@nav13n 
Hi, the PR is looks Ok for me and I'm going to test it this week. Could you also try to use this PR as well? Maybe you will have some comments. Thanks.

**Comment by azhavoro** [comment]
@Langhalsdino Hi, I've tried to follow the README.md, some comments:

1. `--build-arg USER="django" --build-arg DJANGO_CONFIGURATION=production` should be passed to build the backend image as well
2. Could you please specify the k8s version that should be used, I've tried with 1.19 and looks like that configs have some deprecated APIs https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/. 
3. I got the following error:
```sh
kubectl get pods --namespace cvat
NAME                             READY   STATUS             RESTARTS   AGE
cvat-backend-74796d758c-8xm46    0/1     Pending            0          52m
cvat-frontend-7c6cbb6447-rvgqt   1/1     Running            0          52m
cvat-nginx-8669c5658c-rn8j5      0/1     CrashLoopBackOff   15         52m
cvat-postgres-5d596f9c4b-s75bn   0/1     Pending            0          52m
cvat-redis-75cb4864d8-7zk76      1/1     Running            0          52m
```
```sh
kubectl logs cvat-nginx-8669c5658c-rn8j5 --namespace cvat
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: error: /etc/nginx/conf.d/default.conf is not a file or does not exist
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
nginx: [emerg] host not found in upstream "cvat-backend-service" in /etc/nginx/cvat.d/cvat.conf:16
```

**Comment by turowicz** [comment]
Can we get this one completed?

**Comment by BarthV** [comment]
@Langhalsdino It seems that you're using obsoleted api endpoints.
Please use more up2date object version like `apiVersion: apps/v1` for Deployments for example.

**Comment by tamademicheli** [comment]
Hi, i just cloned your work. What you are doing is something different than a template. I used your truly valuable work to build a helm chart, which is working on cpu and without further binding to nuclio. I removed  some part which were not necessary to be declared like redis, postgres and nuclio which i pull them as a chart dependency. The result is a chart with 2 deployments, frontend and backend.

Is someone interested on us to publish a helm chart? if yes where would be the most suitable repo to do that ? I don't like code and chart on the same repo.

**Comment by frankmanbb** [comment]
> Hi, i just cloned your work. What you are doing is something different than a template. I used your truly valuable work to build a helm chart (working with k8s 16+ and the latest endpoints), which is working on cpu and without further binding to nuclio. I removed some part which were not necessary to be declared like redis, postgres and nuclio which i pull them as a chart dependency. The result is a chart with 2 deployments, frontend and backend.
> 
> Is someone interested on us to publish a helm chart? if yes where would be the most suitable repo to do that ? I don't like code and chart on the same repo.

I do have difficulties in applying the original yaml files, which needs many changes and currently still doesn't work for me. so if you can share the helm chart, that will be awesome!

**Comment by tamademicheli** [comment]
> > Hi, i just cloned your work. What you are doing is something different than a template. I used your truly valuable work to build a helm chart (working with k8s 16+ and the latest endpoints), which is working on cpu and without further binding to nuclio. I removed some part which were not necessary to be declared like redis, postgres and nuclio which i pull them as a chart dependency. The result is a chart with 2 deployments, frontend and backend.
> > Is someone interested on us to publish a helm chart? if yes where would be the most suitable repo to do that ? I don't like code and chart on the same repo.
> 
> I do have difficulties in applying the original yaml files, which needs many changes and currently still doesn't work for me. so if you can share the helm chart, that will be awesome!

Hi I will try to create a PR this week. i have to write the README.md and add some script to install kind k8s sig with the ingress controllers.

**Comment by mthogan** [comment]
> > > Hi, i just cloned your work. What you are doing is something different than a template. I used your truly valuable work to build a helm chart (working with k8s 16+ and the latest endpoints), which is working on cpu and without further binding to nuclio. I removed some part which were not necessary to be declared like redis, postgres and nuclio which i pull them as a chart dependency. The result is a chart with 2 deployments, frontend and backend.
> > > Is someone interested on us to publish a helm chart? if yes where would be the most suitable repo to do that ? I don't like code and chart on the same repo.
> > 
> > 
> > I do have difficulties in applying the original yaml files, which needs many changes and currently still doesn't work for me. so if you can share the helm chart, that will be awesome!
> 
> Hi I will try to create a PR this week. i have to write the README.md and add some script to install kind k8s sig with the ingress controllers.

Any update on this? I had plans to leverage the manifests provided in this PR and create a Helm chart as well, but would love to re-use your submission. I am happy to assist with any missing elements if needed. All contributions are much appreciated!

**Comment by Langhalsdino** [comment]
I am really sorry didn't responding to your request. Based on your previous messages i changed a couple of things:

- Merged the current `develop` branch into the PR.
- Fixed the kubernetes API, such that the latest kubernetes version v1.19.3 is supported and tested it for v1.19.3 and 1.9
- Added changelog entry.

This should cover the above mentioned discussions.

**Comment by Langhalsdino** [comment]
@azhavoro and @nmanovic could you do a new code review of this PR? I am really sorry I kept you waiting.

**Comment by nmanovic** [comment]
@azhavoro , need to comment or merge ASAP.

**Comment by Langhalsdino** [comment]
@azhavoro If you need some assistance to setup a local dev environment or other questions, i am happy to assist.

File: kubernetes-templates/04_cvat_backend_deployment.yml
Comment (azhavoro) [inline kubernetes-templates/04_cvat_backend_deployment.yml:39]: Currently, the readme file does not mention this parameter, so we will set the `*` here as in the compose config: https://github.com/openvinotoolkit/cvat/blob/develop/docker-compose.yml#L42.
or please note about that in Readme.md

File: kubernetes-templates/05_cvat_proxy_service.yml
Comment (azhavoro) [inline kubernetes-templates/05_cvat_proxy_service.yml:17]: Do you think reasonable to specify a fixed nodePort as well? Also I think the mention in Readme.md on how to access the deployed CVAT might be helpful.

File: kubernetes-templates/04_cvat_frontend_deployment.yml
Comment (azhavoro) [inline kubernetes-templates/04_cvat_frontend_deployment.yml:33]: These variables are no longer needed and have no effect.

**Comment by azhavoro** [comment]
@Langhalsdino Great job! I tested the PR and it works for me with minor changes (my comments above).
Thank you for the contribution!

**Comment by turowicz** [comment]
Does nuclio work with Kubernetes >= 1.19.x?

**Comment by Langhalsdino** [comment]
> Does nuclio work with Kubernetes >= 1.19.x?
> 
> @Langhalsdino recent commit mentioned it should work with 1.19.x.

I tested the template with 1.9 and 1.19 and therefore deduce that everything between these versions should work. I will update the README.md to include the feedback by @azhavoro this evening.

**Comment by turowicz** [comment]
I wasn't able to get nuclio to work with Kubernetes >1.19 because it does not support Containerd. How did you host model containers without nuclio trying to build them?

File: kubernetes-templates/04_cvat_backend_deployment.yml
Comment (Langhalsdino) [reply kubernetes-templates/04_cvat_backend_deployment.yml]: Adjusted in 590d7f09

File: kubernetes-templates/04_cvat_frontend_deployment.yml
Comment (Langhalsdino) [reply kubernetes-templates/04_cvat_frontend_deployment.yml]: Removed these lines in 590d7f0

File: kubernetes-templates/05_cvat_proxy_service.yml
Comment (Langhalsdino) [reply kubernetes-templates/05_cvat_proxy_service.yml:17]: Since the service is of type nodePort, i am not sure

I added the following to the section to the README.md in 34467324

### Expose the deployment
The service `cvat-proxy-service` is the accesspoint to the deployment.
In order to expose this resource an ingress might be handy [kubernetes ingress documentation](https://kubernetes.io/docs/concepts/services-networking/ingress/).

For debugging puposes it is usefull to forward this service to a port on your localhost.
In the following example `8080` will be used for this purpose [localhost:8080](http://localhost:8080).

```bash
kubectl port-forward service/cvat-proxy-service -n cvat 8080:80
```

**Hint:**
If you are developing locally it might be usefull to replace `{MY_SERVER_URL_COM}` with `localhost`,
such that `/etc/hosts` does not need to override the DNS.

Comment (Langhalsdino) [reply kubernetes-templates/05_cvat_proxy_service.yml:17]: Does this cover your request? Since the `cvat-proxy-service` is already of type NodePort, i am not sure what further change you are suggesting.

Please resolve this conversation if it covers your request.

**Comment by Langhalsdino** [comment]
> I wasn't able to get nuclio to work with Kubernetes >1.19 because it does not support Containerd. How did you host model containers without nuclio trying to build them?

Sorry, i did not phrase this correctly, therefore this misunderstanding is my fault. 

I tested the deployment (part of this PR) with Kubernetes 1.19 and 1.9. Since CVAT works without nuclio this was fine for kubernetes 1.19. Since kubernetes change of using containered was around 1.18, i was suggesting to downgrade your cluster to 1.17, if you would like to use nuclio or wait for nuclio to have fixed this issue.

**Comment by azhavoro** [comment]
LGTM
@ActiveChooN could you please review as well?

**Comment by tamademicheli** [comment]
actually no, sorry i have the pr almost ready but no time for that. don't
wait for me, when i will have time i will see if my pr is still valid.

**Comment by ActiveChooN** [comment]
@azhavoro, LGTM

**Review by azhavoro — APPROVED** [review]

**Review by nmanovic — APPROVED** [review]
@Langhalsdino , thanks for the great contribution!

**Comment by ActiveChooN** [comment]
@Langhalsdino, hi, we've just merged #3102 helm charts, which can easily replace raw k8s templates. Is there any special purpose for k8s templates or its can be removed?

**Comment by jeffliao888** [comment]
I following the document and deploy image: openvino/cvat_server:v1.7.0 in my Kubernetes v1.23.8 , and it seems work fine, but when I changed the image: openvino/cvat_server:v2.x.0 , then it showing "Could not check authorization on the server" in the brower.

Can someone help me if you can success to upgrade the CVAT server image to v2.x.0 and work fine in K8s ? Thank you !

**Pull request merged**
